### PR TITLE
Feature/log status of message

### PR DESCRIPTION
**Pull request by dwybourn**

### CODE REVIEW

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessorTest.java [hunk 1]
```diff
@@ -84,5 +95,35 @@ void shouldThrowIllegalArgumentExceptionForMessageWithNoProcessor() {
...
+    void shouldLogMessageProcessResults() {
+        Message cardMessage = MessageTestDataFactory.aValidMessageWithType(CREATE_NEW_CARD);
+        lenient().when(messageRepository.findAllMessagesByTypeOrderedByDate(CREATE_NEW_CARD)).thenReturn(asList(cardMessage, cardMessage));
+        lenient().when(messageRepository.findAllMessagesByTypeOrderedByDate(MAKE_FIRST_PAYMENT)).thenReturn(emptyList());
+        lenient().when(messageRepository.findAllMessagesByTypeOrderedByDate(SEND_FIRST_EMAIL)).thenReturn(emptyList());
+        //When
+        messageProcessor.processAllMessages();
+        //Then
+        List<ILoggingEvent> events = TestAppender.getEvents();
+        assertThat(events.size()).isEqualTo(1);
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessorTest.java]: Minor, I prefer to use `assertThat(events).hasSize(1)`.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/TestAppender.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package uk.gov.dhsc.htbhf.claimant.testsupport;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.AppenderBase;
+import lombok.Getter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TestAppender extends AppenderBase<ILoggingEvent> {
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/TestAppender.java]: Might it be worthwhile putting this into htbhf-common-test now that we're reusing it?

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java [hunk 3]
```diff
@@ -41,9 +46,26 @@ private void processMessagesOfType(MessageType messageType) {
...
+                .map(messageTypeProcessor::processMessage)
+                .peek(messageStatus -> checkForNullMessageStatus(messageTypeProcessor, messageStatus))
+                .filter(Objects::nonNull)
+                .collect(groupingBy(identity(), counting()));
+
+        statuses.forEach((messageStatus, count) -> log.info("Processed {} {} with status {}",
+                count, count == 1 ? "message" : "messages", messageStatus.name()));
+    }
+
+    private void checkForNullMessageStatus(MessageTypeProcessor messageTypeProcessor, MessageStatus messageStatus) {
```
Comment (gethyn1) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java]: Assuming this is not going to do any more than log a message `logNullMessageStatus` might be a better name.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java [hunk 4]
```diff
@@ -34,16 +40,41 @@ public void processAllMessages() {
...
+                        + messageType + ", there are " + messages.size() + " message(s) in the queue");
             }
-            allMessagesOfType.forEach(messageTypeProcessor::processMessage);
+
+            processMessages(messageTypeProcessor, messages);
+        }
+    }
+
+    private void processMessages(MessageTypeProcessor messageTypeProcessor, List<Message> messages) {
+        log.info("Processing {} {} of type {}", messages.size(), messages.size() == 1 ? "message" : "messages", messageTypeProcessor.supportsMessageType());
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java]: Can we have `messages.size() == 1 ? "message" : "messages"` either out as a function, or simply replaced with `message(s)`, I'd be happy with either. You've also done the same in the log message below too.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessorTest.java (hunk 1 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessorTest.java]: Changed to hasSize.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/TestAppender.java (hunk 2 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/TestAppender.java]: Done.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java (hunk 3 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java]: Done.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java (hunk 4 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java]: Updated to message(s)

**Pull request merged**